util/src/main/java/com/spotify/hamcrest/util/DescriptionUtils.java

```diff
@@ -39,4 +46,84 @@ public static void indentDescription(Description description, Description innerD
     final String indentedLines = INDENTED_LINE_JOINER.join(lines);
     description.appendText(indentedLines).appendText("\n");
   }
+
+  /**
+   * Describes a nested mismatch, useful for nested types like Objects or Maps
+   *
+   * <p>This will print <b>all</b> mismatches occurring in a mismatch list, and properly
+   * handle ellipsis (...). Order will also be maintained based on the order of the allKeys
+   * set. To maintian input order, consider using {@link java.util.LinkedHashSet}
+   * or {@link java.util.LinkedHashMap}
+   *
+   * <p>This will also handle proper indentation in the case of nesting.
+   * Description will contain output that looks like,
+   * <pre>
+   *   {@code
+   * {
+   *   ...
+   *   myKey: expected 1 but was 2
```

Comment:
I find it to be inconsistent that these keys are not quoted, but I suppose that's the nature of `String.toString`. Maybe consider adding `<>` similar to the `is()` matcher?